CHANGELOG/CHANGELOG-1.25.md

```diff
@@ -14,3 +14,5 @@ Changelog for the K8ssandra Operator, new PRs should update the `unreleased` sec
 When cutting a new release, update the `unreleased` heading to the tag being generated and date, like `## vX.Y.Z - YYYY-MM-DD` and create a new placeholder section for  `unreleased` entries.
 
 ## unreleased
+
+* [BUGFIX] [#1572](https://github.com/k8ssandra/k8ssandra-operator/issues/1572) Prevent K8ssandraCluster deletion until the CassDCs have been deleted effectively
```

Comment:
Didn't we discuss that we should add a finalizer to the CassandraDatacenter object?

Reply:
I'm not sure what you mean by that. All of our objects (k8c and cassdc) have finalizers. Do you want to add a second finalizer on the cassdc?
What would be the purpose compared to what's done here?

Reply:
But we don't have finalizer on CassDc set by k8ssandra-operator?

Reply:
Right, let me add this and adjust the deletion code accordingly.